### PR TITLE
feat: added batch

### PR DESCRIPTION
**Pull request by theodorewahle**

### CODE REVIEW

File: smartcar/api.py [hunk 1]
```diff
@@ -59,6 +59,30 @@ def action(self, endpoint, action, **kwargs):
...
+            Response: response from the request to the Smartcar API
+
+        """
+        endpoint = 'batch'
+        url = self._format(endpoint)
+        json = dict()
+        json["headers"] = { "sc-unit-system" : self.unit_system }
+        json["requests"] = requests
+
+        print("HERE's BATCH")
```
Comment (theodorewahle) [inline smartcar/api.py]:
```suggestion
```

File: smartcar/api.py [hunk 2]
```diff
@@ -59,6 +59,30 @@ def action(self, endpoint, action, **kwargs):
...
+        print(url)
```
Comment (theodorewahle) [inline smartcar/api.py]:
```suggestion
```

File: smartcar/api.py [hunk 3]
```diff
@@ -59,6 +59,30 @@ def action(self, endpoint, action, **kwargs):
...
+        print(json)
```
Comment (theodorewahle) [inline smartcar/api.py]:
```suggestion
```

File: smartcar/api.py [hunk 4]
```diff
@@ -59,6 +59,30 @@ def action(self, endpoint, action, **kwargs):
...
+        print(self.auth)
```
Comment (theodorewahle) [inline smartcar/api.py]:
```suggestion
```

File: smartcar/api.py [hunk 5]
```diff
@@ -59,6 +59,24 @@ def action(self, endpoint, action, **kwargs):
 
         return requester.call('POST', url, json=json, headers=self.auth)
 
+    def batch(self, requests):
+        """ Sends POST requests to Smartcar API
+
+        Args:
+            requests (object[]) - an array of objects containing a pat
```
Comment (theodorewahle) [inline smartcar/api.py]:
```suggestion
            requests (object[]) - an array of objects containing a 'path' key
```

File: tests/test_vehicle.py [hunk 6]
```diff
@@ -279,3 +279,58 @@ def test_unlock(self):
...
+
+        requests = []
+        for path in endpoints:
+            requests.append({ "path" : path })
+
+        request_body = dict()
+        request_body['headers'] = headers
+        request_body['requests'] = requests
+
+        responses = [
```
Comment (adolfoportilla) [inline tests/test_vehicle.py]:
```suggestion
        expected_response = {
        "responses": [
```

File: tests/test_vehicle.py [hunk 7]
```diff
@@ -279,3 +279,58 @@ def test_unlock(self):
...
+                "headers": headers,
+                "path": '/sunroof',
+                "code": 501,
+                "body": {
+                    "error": 'vehicle_not_capable_error',
+                    "message": 'Vehicle is not capable of performing request.'
+                }
+            }
+        ]
+        data = { "responses" : responses }
```
Comment (adolfoportilla) [inline tests/test_vehicle.py]:
```suggestion
```

File: tests/test_vehicle.py [hunk 8]
```diff
@@ -279,3 +279,58 @@ def test_unlock(self):
...
+        self.queue('POST', 'batch', body=data)
+        response = self.vehicle.batch(endpoints)
+        self.assertEqual(response['responses'], responses)
```
Comment (adolfoportilla) [inline tests/test_vehicle.py]:
```suggestion
        self.assertEqual(response['responses'], expected_response.responses)
```

File: tests/test_vehicle.py [hunk 9]
```diff
@@ -279,3 +279,58 @@ def test_unlock(self):
...
+                "path": '/sunroof',
+                "code": 501,
+                "body": {
+                    "error": 'vehicle_not_capable_error',
+                    "message": 'Vehicle is not capable of performing request.'
+                }
+            }
+        ]
+        data = { "responses" : responses }
+        self.queue('POST', 'batch', body=data)
```
Comment (adolfoportilla) [inline tests/test_vehicle.py]: Add some spaces to following AAA that way its easier to read

File: tests/tests_e2e/test_vehicle.py [hunk 10]
```diff
@@ -75,6 +75,9 @@ def test_unlock(self):
         unlock = self.vehicle.unlock()
         self.assertEqual(unlock["status"], "success")
 
+    def test_batch(self):
+        batch_responses = self.vehicle.batch(['/odometer'])
```
Comment (adolfoportilla) [inline tests/tests_e2e/test_vehicle.py]:
```suggestion
        batch_responses = self.vehicle.batch(['/odometer', '/location'])
```

File: smartcar/api.py [hunk 11]
```diff
@@ -59,6 +59,24 @@ def action(self, endpoint, action, **kwargs):
...
+
+        Returns:
+            Response: response from the request to the Smartcar API
+
+        """
+        endpoint = 'batch'
+        url = self._format(endpoint)
+        json = dict()
+        json["headers"] = { "sc-unit-system" : self.unit_system }
+        json["requests"] = requests
```
Comment (adolfoportilla) [inline smartcar/api.py]: Simpler?
```
json = {
    "headers": { "sc-unit-system" : self.unit_system },
    "requests": requests
}
```

Comment (theodorewahle) [reply smartcar/api.py]: yes

File: tests/test_vehicle.py (hunk 9 above)
Comment (theodorewahle) [reply tests/test_vehicle.py]: done! great catch

File: smartcar/__init__.py [hunk 12]
```diff
@@ -1,4 +1,4 @@
-__version__ = '4.1.5'
+__version__ = '4.1.6'
```
Comment (theodorewahle) [inline smartcar/__init__.py]:
```suggestion
__version__ = '4.2.0'
```

**Pull request merged**